lib/ref_arch_setup/install.rb

```diff
@@ -70,6 +78,223 @@ def handle_pe_conf(pe_conf_path)
+      puts
+
+      success = download_pe_tarball(url, "localhost")
+      success = upload_pe_tarball("#{TMP_WORK_DIR}/#{filename}", target_master) if success
+      return success
+    end
+
+    # Downloads the specified PE tarball URL by either downloading directly to the
+    # target master or downloading locally and copying to the target master
+    #
+    # @author Bill Claytor
+    #
+    # @param [string] url The pe tarball URL
+    # @param [string] target_master Host to install on
+    #
+    # @return [true,false] Based on exit status of the bolt task
+    def handle_tarball_url(url, target_master)
+      uri = URI.parse(url)
+      filename = File.basename(uri.path)
+
+      # TODO: improve check for localhost
+      if target_master.equal?("localhost")
+
+        # destination = default
+        success = download_pe_tarball(url, target_master)
+        raise "Failed downloading #{url} to #{target_master}" unless success
+
+      else
+
+        puts "Specified target master #{target_master} is not localhost"
+        puts
+
+        success = download_pe_tarball(url, target_master)
+
+        # if at first you don't succeed...
+        puts "Failed download attempt to #{target_master}." unless success
+        success = download_and_move_pe_tarball(url, filename, target_master) unless success
+
+        raise "Failed downloading #{url} to localhost and moving to #{target_master}" unless success
+
+      end
+
+      return filename
+    end
+
+    # Copies the PE tarball from the specified location to the temp working directory
+    # on the target master
+    #
+    # @author Bill Claytor
+    #
+    # @param [string] pe_tarball_path The pe tarball path
+    # @param [string] target_master Host to install on
+    #
+    # @return [true,false] Based on exit status of the bolt task
+    def copy_pe_tarball(pe_tarball_path, target_master)
+      filename = File.basename(pe_tarball_path)
+      command = "cp #{pe_tarball_path} #{TMP_WORK_DIR}/#{filename}"
+      success = BoltHelper.run_cmd_with_bolt(command, target_master)
+      return success
+    end
+
+    # Handles the specified tarball path when the target master is not localhost
+    #
+    # @author Bill Claytor
+    #
+    # @param [string] pe_tarball_path The pe tarball path
+    # @param [string] target_master Host to install on
+    #
+    # @return [true,false] Based on exit status of the bolt task
+    def handle_tarball_with_remote_target_master(pe_tarball_path, target_master)
+      remote_flag = "#{target_master}:"
+
+      if pe_tarball_path.start_with?(remote_flag)
+        tarball_path = pe_tarball_path.sub!(remote_flag, "")
+        success = file_exist_on_target_master?(tarball_path, target_master)
+        success = copy_pe_tarball(tarball_path, target_master) if success
+      else
+        success = File.exist?(pe_tarball_path)
+        success = upload_pe_tarball(pe_tarball_path) if success
```

Comment:
this is another case where target host is an input to the method, but isn't passed to the upload method...

Reply:
Will update to use `@target_master`